[10.0][MIG] : account_invoice_customer_ref_unique

misern2 opened the pull request:
# USAGE

This module adds a insensitive constraint on the name Customer Reference
(name field): (partner_id, name) must be unique.

misern2 committed:
[MIG] v10 migration

* Test coverage added
* Spanish translation

misern2 committed:
Travis build test

Travis Build Test

misern2 committed:
flake errors

flake errors

misern2 committed:
Falke modifications

flake modifications

misern2 committed:
Readme.rst modifications

AaronHForgeFlow reviewed (approved):
:+1: 
It'll be great a pre-install hook to prevent this constraint to be triggered when installing.

moylop260 commented on account_invoice_customer_ref_unique/models/account_invoice.py:
```diff
@@ -19,47 +19,34 @@
-            invoice = self.browse(cr, uid, i_id, context=context)
+    @api.multi
+    @api.constrains('name')
+    def _check_unique_name_insensitive(self):
```
In order to increase the performance is faster use a sql constraint based on computed fields with store=True

moylop260 replied:
check the following example where is 4x faster than a python constraint using too many invoices https://github.com/Vauxoo/addons-vauxoo/blob/8.0/supplier_invoice_number_unique/model/account_invoice.py

misern2 committed:
[FIX] comercial_partner_id constraint
[FIX] minor travis build issues
[FIX] Test updated : Integrity Error

misern2 commented:
@moylop260 Thanks for your revision. I pushed a new commit with the SQL Constraints

moylop260 commented on account_invoice_customer_ref_unique/models/account_invoice.py:
```diff
@@ -0,0 +1,46 @@
+                invoice.customer_ref_strip = invoice.name.lower()
+    _sql_constraints = [
+        ('unique_customer_ref_strip', 'UNIQUE('
+         'customer_ref_strip, company_id, partner_id)',
```
s/partner_id/commercial_partner_id/g` in order to supports the case where other address is used for the same partner

moylop260 commented on .travis.yml:
```diff
@@ -1,7 +1,7 @@
 
 python:
-  - "2.7"
+  - "2.7"   
```
remove this small change

moylop260 reviewed (changes requested):
LGTM just small changes.

And check our beta message of the lint build:
https://travis-ci.org/OCA/account-invoicing/jobs/281607391#L743

```bash
************* Module account_invoice_customer_ref_unique
account_invoice_customer_ref_unique/README.rst:32: [E7901(rst-syntax-error), ]  Unexpected possible title overline or transition.|Treating it as ordinary text because it's so short.
account_invoice_customer_ref_unique/README.rst:1: [W7908(missing-newline-extrafiles), ]  Missing newline
************* Module account_invoice_customer_ref_unique.__manifest__
account_invoice_customer_ref_unique/__manifest__.py:1: [C8201(no-utf8-coding-comment), ] No UTF-8 coding comment found: Use `# coding: utf-8` or `# -*- coding: utf-8 -*-`
account_invoice_customer_ref_unique/__manifest__.py:23: [C8103(manifest-deprecated-key), ] Deprecated key "description" in manifest file
************* Module account_invoice_customer_ref_unique.models.account_invoice
account_invoice_customer_ref_unique/models/account_invoice.py:1: [C8201(no-utf8-coding-comment), ] No UTF-8 coding comment found: Use `# coding: utf-8` or `# -*- coding: utf-8 -*-`
```

moylop260 commented:
...and create a hook to auto-fix all current wrong info breaking the constraint:
https://travis-ci.org/OCA/account-invoicing/jobs/281607395#L1407

misern2 commented:
@moylop260 I think it's all finshed. I also checked the travis log file and now  there is not any error beta message.
Thanks.

moylop260 commented on account_invoice_customer_ref_unique/__init__.py:
```diff
@@ -20,4 +20,5 @@
 
-from . import account_invoice
+
+from . import models
```
Missing newline

moylop260 commented on account_invoice_customer_ref_unique/__init__.py:
```diff
@@ -20,4 +20,5 @@
 #
 ##############################################################################
```
Use shortest license like as template module:
https://github.com/OCA/maintainer-tools/blob/master/template/module/models/model_name.py#L2

moylop260 commented on account_invoice_customer_ref_unique/README.rst:
```diff
@@ -0,0 +1,73 @@
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit https://odoo-community.org.
```
missing newline

moylop260 commented on account_invoice_customer_ref_unique/README.rst:
```diff
@@ -0,0 +1,73 @@
+
+Configuration
+=============
+
```
Remove unused sections

moylop260 commented on account_invoice_customer_ref_unique/models/account_invoice.py:
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# #############################################################################
```
Short license

moylop260 reviewed (changes requested):
Sorry I didn't see this before...

Thanks for your great work!

houssine78 commented on account_invoice_customer_ref_unique/models/account_invoice.py:
```diff
@@ -0,0 +1,46 @@
+    def _compute_customer_ref_strip(self):
+        for invoice in self:
+            invoice_type = invoice.type
+            if invoice_type not in ['out_invoice', 'out_refund']:
```
maybe we can remove the previous line as the invoice_type variable is only used once in the if statement

misern2 committed:
Reviews Fixes

shorter license updated
Removed unused sections in README
removed invoice_type variable in account_invoice.py

misern2 commented:
Changes requested.
The travis build on my branch was Ok and It fails on this PR in my last commit. Am I missing something ?

pedrobaeza commented:
Travis is red

misern2 committed:
Test fixed
